src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -215,61 +221,140 @@ private void turnOffReadiness(Throwable exception){
+                    // this code executed in executor service and exception thrown here will not be handled anywhere
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
                 }
+            })
+        );
+    }
 
-                @Override
-                public void handleRecoveryStarted(Recoverable recoverable) {
+    private void addShutdownListenerToConnection(Connection conn) {
+        conn.addShutdownListener(cause -> {
+            LOGGER.debug("Closing the connection: ", cause);
+            if (cause.isHardError() && cause.getReference() instanceof Connection) {
+                Connection connectionCause = (Connection) cause.getReference();
+                Method reason = cause.getReason();
+                if (reason instanceof AMQImpl.Connection.Close) {
+                    var castedReason = (AMQImpl.Connection.Close) reason;
+                    if (castedReason.getReplyCode() != 200) {
+                        StringBuilder errorBuilder = new StringBuilder("RabbitMQ hard error occupied: ");
+                        castedReason.appendArgumentDebugStringTo(errorBuilder);
+                        errorBuilder.append(" on connection ");
+                        errorBuilder.append(connectionCause);
+                        LOGGER.warn(errorBuilder.toString());
+                    }
                 }
-            };
+            }
+        });
+    }
+
+    private void addRecoveryListenerToConnection(Connection conn) {
+        if (conn instanceof Recoverable) {
+            Recoverable recoverableConnection = (Recoverable) conn;
             recoverableConnection.addRecoveryListener(recoveryListener);
             LOGGER.debug("Recovery listener was added to connection.");
         } else {
             throw new IllegalStateException("Connection does not implement Recoverable. Can not add RecoveryListener to it");
         }
     }
 
+    private void addBlockedListenersToConnection(Connection conn) {
+        conn.addBlockedListener(new BlockedListener() {
+            @Override
+            public void handleBlocked(String reason) {
+                LOGGER.warn("RabbitMQ blocked connection: {}", reason);
+            }
+
+            @Override
+            public void handleUnblocked() {
+                LOGGER.warn("RabbitMQ unblocked connection");
+            }
+        });
+    }
+
     public boolean isOpen() {
         return connection.isOpen() && !connectionIsClosed.get();
     }
 
     @Override
-    public void close() {
+    public void close()  throws IOException {
         if (connectionIsClosed.getAndSet(true)) {
             LOGGER.info("Connection manager already closed");
             return;
         }
 
         LOGGER.info("Closing connection manager");
 
+        for (ChannelHolder channelHolder: channelsByPin.values()) {
+            channelHolder.channel.abort();
```

Comment:
We doesn't execute the code below if this instruction throws the IOException.
Maybe collect all exceptions and throw an aggregated exception with the thrown exceptions as suppressed

Reply:
I think we should just log exceptions the same as for channel.close(). Currently ConnectionManager::close does not throws IOException, so I just catch and log it.